src/journal-remote/journal-remote-parse.c

```diff
@@ -441,9 +441,47 @@ static int process_data(RemoteSource *source) {
         }
 }
 
+/*
+ * Returns 1 if field _SOURCE_MONOTONIC_TIMESTAMP already exists in the log entry,
+ * 0 if it does not.*/
+int source_monotonic_timestamp_exists(RemoteSource *source)
+{
+    size_t i;
+    size_t tag_len = sizeof("_SOURCE_MONOTONIC_TIMESTAMP=") - 1;
+    int ret = -1;
+
+    for (i=0;i<source->iovw.count; i++)
+    {
+        if (source->iovw.iovec[i].iov_len >= tag_len)
+        {
+            ret = memcmp(source->iovw.iovec[i].iov_base, "_SOURCE_MONOTONIC_TIMESTAMP=", tag_len);
+            if (ret==0)
+                break;
+        }
+    }
+    return ret==0 ? 1 : 0;
+}
```

Comment:
Please keep to the coding style implemented by other files in the source tree which is also described in `CODING_STYLE`. In particular, `{` are not put on a new line, we use spaces around `==`, before `*` in casts and after `;` in  `for` etc.